homeassistant/components/growatt_server/sensor.py

```diff
@@ -257,6 +258,45 @@ def update(self):
         except json.decoder.JSONDecodeError:
             _LOGGER.error("Unable to fetch data from Growatt server")
 
-    def get_data(self, variable):
+    def get_currency(self):
+        """Get the currency."""
+        return self.data.get("currency")
+
+    def get_data(self, entity_description):
         """Get the data."""
-        return self.data.get(variable)
+        variable = entity_description.api_key
+        api_value = self.data.get(variable)
+        previous_value = self.previous_values.get(variable, None)
+        return_value = api_value
+
+        # If we have a 'drop threshold' specified, then check it and correct if needed
+        if (
+            entity_description.previous_value_drop_threshold is not None
+            and previous_value is not None
+        ):
+            _LOGGER.debug(
+                "%s - Drop threshold specified (%s), checking for drop... API Value: %s, Previous Value: %s",
+                entity_description.name,
+                entity_description.previous_value_drop_threshold,
+                api_value,
+                previous_value,
+            )
+            diff = float(api_value) - float(previous_value)
+
+            # Check if the value has dropped (negative value i.e. < 0) and it has only dropped by a small amount, if so, use the previous value
+            # Note - The energy dashboard takes care of drops within 10% of the current value, however if the value is low e.g. 0.2 and drops by 0.1 it classes as a reset.
```

Comment:
Please break long comments and strings around 88 characters per line.

Reply:
Addressed in new commit